services/utils.js

```diff
@@ -0,0 +1,75 @@
+const bcrypt = require("bcrypt");
+const EMAIL_REGEX =
+  /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+const isValidEmailSyntax = (email_address) => EMAIL_REGEX.test(email_address);
+
+const isValidPassword = (password) =>
+  password.length > 8 &&
+  password.length < 20 &&
+  /[A-Z]/.test(password) &&
+  /[a-z]/.test(password) &&
+  /\d/.test(password) &&
+  /[!@#$%^&*()_+\-=[\]{};':"\\|,.<>/?]/.test(password);
+
+const composite = (base, salt, pepper) => `${salt}${base}${pepper} `;
+
+const compare = async (storedPassword, userSalt, input, RANGE) => {
+  for (let i = 0; i < RANGE; i++) {
+    let cmp_res = await bcrypt.compare(
+      composite(input, userSalt, i),
+      storedPassword
+    );
+    if (cmp_res) return true;
+  }
+  return false;
+};
+
+const getRandString = () => `${Math.random()}${Date.now()}`;
+
+const WEIGHT = "weight";
+const DESTINATION = "destination";
+const validFields = [WEIGHT, DESTINATION];
+
+const objIsEdge = (obj, vertexIndex) => {
+  const fields = Object.keys(obj);
+  if (fields.length != validFields.length) return false;
+  for (let i = 0; i < fields.length; i++) {
+    if (!validFields.includes(fields[i])) return false;
+  }
+  if (
+    !fields.includes(DESTINATION) ||
+    obj[DESTINATION] === null ||
+    obj[DESTINATION] == `${vertexIndex}`
+  )
+    return false;
```

Comment:
You can extract this really long condition into a variable (or a couple of them). 
And then in the if just do something like this `if(condition2 || condition2...)`